logger/src/lib.rs

```diff
@@ -54,6 +57,27 @@ pub fn setup() {
     setup_with_default("error");
 }
 
+// When we have multi-threading tests like local-clusters, it's nice to print
+// the thread id at the beginning, so you can filter out logs belonging to
+// different threads.
+pub fn setup_logging_with_thread_id() {
+    env_logger::Builder::from_default_env()
+        .format(|buf, record| {
+            writeln!(
+                buf,
+                "{:?} [{} {} {}:{}] {}",
+                std::thread::current().id(),
+                buf.timestamp_nanos(),
+                buf.default_level_style(record.level())
+                    .value(record.level()),
+                record.target(),
+                record.line().unwrap_or(0),
+                record.args()
```

Comment:
Unfortunate we don't have access to `env_logger`'s `DefaultFormat`, would be nice to just append the thread id to that format instead of recreating it here 😢

Reply:
That's probably a neater solution, this is the best I can figure out though, and we definitely need the thread id in multi-thread tests so we don't need to patch pubkey here and there for debugging any more.